### PR TITLE
Fixes #2464

### LINKED ISSUE #2464: The character ÿ has begun appearing before compartment names in event notices.
#### Description of issue
Extremely minor but I figured I'd report it anyway. A strange character has started sneaking in before compartment names in notices. For example:

Bioscans indicate that lizards have been breeding in the ÿChemistry Bay. Further infestation is likely if left unchecked.

Pressure readings indicate an imminent explosion in the ÿMaintenance Corridors disposal systems. Piping sections may be damaged

#### Difference between expected and actual behavior

The character ÿ should not be preceding compartment names.

#### Steps to reproduce

None. Just have to wait for an event to happen.

#### Length of time in which bug has been known to occur

I just noticed this today, 5-28-2022 when testing other issues.

#### Client version, Server revision & Game ID

Client Version: 514
Server Revision: ea3cdbe93b0a70bb9e3d1d6f1bf108c9283c6717 - dev - 2022-05-27
Game ID: cia-avgO
Current map: Tradeship Ivenmoth

#### Issue bingo

- [x] Issue could be reproduced at least once
- [ ] Issue could be reproduced by different players
- [ ] Issue could be reproduced in multiple rounds
- [x] Issue happened in a recent (less than 7 days ago) round
- [x] [Couldn't find an existing issue about this](https://github.com/NebulaSS13/Nebula/issues)

### PR DESCRIPTION
**Pull request by noelle-lavenza**
## Description of changes
- Adds `display_name` to areas, set on init and rename, essentially a cached version of `strip_improper(name)`.
- Renaming an area now allows you to set it as improper, if you begin it with `the `. For example, the area name `Foo` will display as "in Foo" in messages, but the name `the Foo` will display as "in the Foo" in messages and have the display name `Foo`.
- Adds a return type annotation to `get_area`, slight rewrite.

## Why and what will this PR improve
Fixes NebulaSS13/Nebula#2464

## Authorship
Me.

## Changelog
:cl:
bugfix: Fixes ÿ in area names.
/:cl:

### CODE REVIEW

**Comment by MarinaGryphon** [comment]
- [x] Tested

File: code/_helpers/game.dm [hunk 1]
```diff
@@ -17,14 +17,13 @@
...
 
 /proc/get_area(O)
+	RETURN_TYPE(/area)
 	var/turf/loc = get_turf(O)
-	if(loc)
-		var/area/res = loc.loc
-		.= res
+	return loc?.loc
 
 /proc/get_area_name(N) //get area by its name
```
Comment (MistakeNot4892) [inline code/_helpers/game.dm]: Checking actual uses of this, I'm confused about what it's for.

Comment (noelle-lavenza) [reply code/_helpers/game.dm]: Every use but one appears to be incorrect. I can fix those if you'd like.

**Comment by noelle-lavenza** [comment]
image
Tested.

**Comment by noelle-lavenza** [comment]
Nevermind, other stuff broke in a local commit.

**Comment by noelle-lavenza** [comment]
Okay, tested and pushed. Works now.

File: code/game/objects/items/devices/powersink.dm [hunk 2]
```diff
@@ -113,7 +113,7 @@
 				"<span class='notice'>You activate \the [src].</span>",
 				"<span class='italics'>You hear a click.</span>")
 			message_admins("Power sink activated by [key_name_admin(user)] at (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[src.x];Y=[src.y];Z=[src.z]'>JMP</a>)")
-			log_game("Power sink activated by [key_name(user)] at [get_area_name(src)]")
+			log_game("Power sink activated by [key_name(user)] at [get_area(src).name]")
```
Comment (MistakeNot4892) [inline code/game/objects/items/devices/powersink.dm]: Please cast this out.

File: code/game/machinery/camera/camera.dm [hunk 3]
```diff
@@ -80,14 +80,13 @@
...
-		if(isturf(loc))
-			var/area/A = get_area(src)
-			if(A)
-				for(var/obj/machinery/camera/C in A)
-					if(C == src) continue
-					if(C.number)
-						number = max(number, C.number+1)
-				c_tag = "[A.name][number == 1 ? "" : " #[number]"]"
+		var/area/A = get_area(src)
+		if(isturf(loc) && A)
```
Comment (MistakeNot4892) [inline code/game/machinery/camera/camera.dm:84]: Doesn't `A` being truthy imply loc is a turf?

File: code/game/machinery/computer/ai_core.dm [hunk 4]
```diff
@@ -248,7 +248,7 @@ var/global/list/deactivated_ai_cores = list()
 
 	var/list/cores = list()
 	for(var/obj/structure/aicore/deactivated/D in global.deactivated_ai_cores)
-		cores["[D] ([get_area(D)])"] = D
+		cores["[D] ([get_area(D).proper_name])"] = D
```
Comment (MistakeNot4892) [inline code/game/machinery/computer/ai_core.dm]: Please cast this out and check it.

File: code/game/machinery/embedded_controller/docking_program.dm [hunk 5]
```diff
@@ -297,4 +297,4 @@
 		if (STATE_DOCKED) return "docked"
 
 /datum/computer/file/embedded_program/docking/proc/get_name()
-	return display_name ? display_name : "[get_area(master)] ([master.x], [master.y])"
+	return display_name ? display_name : "[get_area(master).proper_name] ([master.x], [master.y])"
```
Comment (MistakeNot4892) [inline code/game/machinery/embedded_controller/docking_program.dm]: Pls cast

File: code/game/machinery/supplybeacon.dm [hunk 6]
```diff
@@ -79,7 +79,7 @@
 	icon_state = "beacon_active"
 	activated = TRUE
 	START_PROCESSING(SSobj, src)
-	admin_attacker_log(user, "has activated \a [src] at [get_area(src)]")
+	admin_attacker_log(user, "has activated \a [src] at [get_area(src).proper_name]")
```
Comment (MistakeNot4892) [inline code/game/machinery/supplybeacon.dm]: Pls cast

File: code/modules/detectivework/tools/sample_kits/_sample.dm [hunk 7]
```diff
@@ -11,7 +11,7 @@
 	if(supplied)
 		copy_evidence(supplied)
 		name = "[initial(name)] (\the [supplied])"
-		object = "[supplied], at [get_area(supplied)]"
+		object = "[supplied], at [get_area(supplied).proper_name]"
```
Comment (MistakeNot4892) [inline code/modules/detectivework/tools/sample_kits/_sample.dm]: Pls cast

File: code/modules/events/blob.dm [hunk 8]
```diff
@@ -13,7 +13,7 @@
 		kill()
 		return
 
-	log_and_message_admins("Blob spawned in \the [get_area(T)]", location = T)
+	log_and_message_admins("Blob spawned in \the [get_area(T).proper_name]", location = T)
```
Comment (MistakeNot4892) [inline code/modules/events/blob.dm]: Pls cast

File: code/game/machinery/camera/camera.dm (hunk 3 above)
Comment (noelle-lavenza) [reply code/game/machinery/camera/camera.dm:84]: Uh, no. The loc being a turf *should* imply that A is truthy, though, but that's the other way around.
get_area works even if the direct loc of the object isn't a turf, which is what isturf(loc) checks.

Comment (MistakeNot4892) [reply code/game/machinery/camera/camera.dm:84]: Oh, yep, I follow.

**Comment by MistakeNot4892** [comment]
Rather than casting I would also consider `get_area(foo)?.proper_name` acceptable.

File: code/_helpers/game.dm [hunk 9]
```diff
@@ -21,7 +21,11 @@
 	var/turf/loc = get_turf(O)
 	return loc?.loc
 
-/proc/get_area_name(N) //get area by its name
+/proc/get_area_name(O) //get area's proper name
+	var/area/A = get_area(O)
+	return A.proper_name
```
Comment (MistakeNot4892) [inline code/_helpers/game.dm]: `A?.proper_name` ?

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**